skrub/_joiner.py

```diff
@@ -45,66 +132,88 @@ class Joiner(TransformerMixin, BaseEstimator):
-    match_score : float, default=0
-        Distance score between the closest matches that will be accepted.
-        In a [0, 1] interval. 1 means that only a perfect match will be
-        accepted, and zero means that the closest match will be accepted,
-        no matter how distant.
-        For numerical joins, this defines the maximum Euclidean distance
-        between the matches.
-    analyzer : {'word', 'char', 'char_wb'}, default=`char_wb`
-        Analyzer parameter for the CountVectorizer used for
-        the string similarities.
-        Describes whether the matrix `V` to factorize should be made of
-        word counts or character n-gram counts.
-        Option `char_wb` creates character n-grams only from text inside word
-        boundaries; n-grams at the edges of words are padded with space.
-    ngram_range : 2-tuple of int, default=(2, 4)
-        The lower and upper boundaries of the range of n-values for different
-         n-grams used in the string similarity. All values of `n` such
-         that ``min_n <= n <= max_n`` will be used.
+    max_dist : float, default=np.inf
+        Maximum acceptable (rescaled) distance between a row in the
+        ``main_table`` and its nearest neighbor in the ``aux_table``. Rows that
+        are farther apart are not considered to match. By default, the distance
+        is rescaled so that a value between 0 and 1 is typically a good choice,
+        although rescaled distances can be greater than 1 for some choices of
+        ``ref_dist``. ``None``, ``"inf"``, ``float("inf")`` or ``numpy.inf``
+        mean that no matches are rejected.
+    ref_dist : reference distance for rescaling, default = 'random_pairs'
+        Options are {"random_pairs", "second_neighbor", "self_join_neighbor",
+        "no_rescaling"}. See above for a description of each option. To
+        facilitate the choice of ``max_dist``, distances between rows in
+        ``main_table`` and their nearest neighbor in ``aux_table`` will be
+        rescaled by this reference distance.
+    string_encoder : scikit-learn transformer used to vectorize text columns
+        By default a ``HashingVectorizer`` combined with a ``TfidfTransformer``
+        is used. Here we use raw TF-IDF features rather than transforming them
+        for example with ``GapEncoder`` or ``MinHashEncoder`` because it is
+        faster, these features are only used to find nearest neighbors and not
+        used by downstream estimators, and distances between TF-IDF vectors
+        have a somewhat simpler interpretation.
+    add_match_info : bool, default=True
+        Insert some columns whose names start with `skrub_Joiner` containing
+        the distance, rescaled distance and whether the rescaled distance is
+        above the threshold. Those values can be helpful for an estimator that
+        uses the joined features, or to inspect the result of the join and set
+        a ``max_dist`` threshold.
```

Comment:
Why should the default be True here? I see that its False in the fuzzy_join, and I think that adding these additional columns is not what the user would want by default.

Reply:
Gael suggested making it True for Joiner and False for fuzzy_join: https://github.com/skrub-data/skrub/pull/821#issuecomment-1822605890

Reply:
Ok, sorry, didn't see this